### DISCUSSION TITLE
Grainy reading font

### DISCUSSION THREAD

**Discussion by Av1ators** [sumatrapdfreader/sumatrapdf#4133] [General]
Comparing various pdf readers, sumatra PDF, displays the characters with a little different font, slightly grainier. is it possible to improve or change it?

**Comment by GitHubRulesOK** [comment]
Vector P{DF fonts will normally be crisp outlines at any scale but pixels on screen will often be anti-aliased to intentionally feathered edges. So it depends on source and screen resolution and zoom etc. 

For there to be any improvement (if possible) a sample would be needed but generally SumatraPDF behaves with default setting to work well for ALL formats in a group without especial handling of a few.

So the issue has been raised before "Why is the same image shown different as say PNG or PDF"? is because PDF says the pixels generally must be hard edges (aliased) whilst PNG expectation is softer anti-aliased and no setting intended for altering from PDF standard or vice versa.

**Comment by Av1ators** [comment]
I am talking about the font only on the pdf. I have installed 4 other pdf reading programs and 2 browsers, and only sumatra has more pixel graining

**Comment by GitHubRulesOK** [comment]
PDF standard is never perfectly the same between apps as open to interpretation but for the case of fonts the statements include.
Elementary objects such as strokes, fills, and text have an intrinsic shape,....NOTE 1
Mathematically, elementary objects have "hard" edges, with a shape value of either 0.0 or 1.0 at every point. However, when such objects are rasterized to device pixels, the shape values along the boundaries can be anti-aliased, taking on fractional values representing fractional coverage of those pixels. When such anti-aliasing is performed, it is important to treat the fractional coverage as shape rather than opacity.

So without anti-alias Acrobat would show standard text like this (not desirable)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/1220926e-d474-4d45-8d97-33c92f9eb69a)



Not going to try to show every example of how that differs in many cases I see, but take that simple description in Acrobat Firefox and SumatraPDF. Acrobat uses "upscaled" (smaller) sharp edges, Firefox is naturally  "fuzzy", SumatraPDF tries to strike a balance in accordance with the standards own stated standard.  
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/19fb8acc-1e68-40e1-a191-f6c3c5ecc697)

**Comment by Av1ators** [comment]
Understood, but is there room for improvement? Or will it stay like this forever?

**Comment by kjk** [comment]
It's subjective and I have no bandwidth to work on that. But if anyone makes an improvement, I'll be happy to review and merge.
